parser/src/function.rs

```diff
@@ -6,9 +6,19 @@ type FunctionArgument = (Option<Option<String>>, ast::Expression);
 pub fn parse_args(func_args: Vec<FunctionArgument>) -> Result<ast::ArgumentList, LexicalError> {
     let mut args = vec![];
     let mut keywords = vec![];
+
+    let mut keyword_names = vec![];
```

Comment:
Using a set like object instead of a vector would gain a speedup for larger amounts of arguments. The `contains` method for `Vec` is likely O(n) while the `contains` method (or equivalent method) on set like types is O(1).

Reply:
Might be worthwhile to use ```<collection>::with_capacity(func_args.len())``` to avoid having lots of small allocations as they get added to the collection.

https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=d1eaab59dcae1ee483e992e19f819c91

Reply:
The vector re-allocation scheme is actually pretty good in rust. The amount of memory is roughly doubled, so capacity will grow something like this: 1, 2, 4, 8, 16 etc.. This results in few allocations.